### ISSUE TITLE
So whats next?

### ISSUE THREAD

**Issue by farkon00** [farkon00/cont#2]
So I kinda run out of ideas. Leave a comment for this issue, if you have any. But please check if idea was already implemented in commits  or in examples.

**Comment by tixvage** [comment]
c interop?

**Comment by farkon00** [comment]
@tixvage Sounds not that bad. Only problem, i dont want to link to libc, but its better to use libc, then write sdl in cont. Thanks for response!

**Comment by farkon00** [comment]
Plans: add different types of fors(like range or null-terminated), make compile time if

**Comment by farkon00** [comment]
> Plans: add different types of fors(like range or null-terminated), make compile time if

Finished.  
I am currently planning to make types in runtime, generics and overloading for procedures

**Comment by lambdasapple** [comment]
I've noticed that you used `18446744073709551615` as -1 in `test.cn`. Not sure if that's old code, but how about supporting negative numbers.

**Comment by farkon00** [comment]
> I've noticed that you used `18446744073709551615` as -1 in `test.cn`. Not sure if that's old code, but how about supporting negative numbers.

That's a pretty old code, and it might not even compile nowadays, I do not recomend using it as a reference. Negative numbers are supported since commit [a02b962](https://github.com/farkon00/cont/commit/a02b962013dd11029c0f2a28220a7e26626f6920)

**Comment by lambdasapple** [comment]
Since cont is a pretty low level language (I mean, around as "low level" as c), how about goto or labeled loops.

**Comment by farkon00** [comment]
> Since cont is a pretty low level language (I mean, around as "low level" as c), how about goto or labeled loops.

They are not implemented for the same reason as `break` and `continue`: diffuculties with type checking in a stack-based language. But in my opinion these keywords for loops would be easier to implement than goto. And because for now we don't have `break` and `continue` I don't see a need for labeled loops. If there are other uses for them I would be glad to know them.

**Comment by Francicoria** [comment]
Function overloading

**Comment by farkon00** [comment]
> Function overloading

I think this should be implemented, but only after I merge in the wat64 branch. Also there might be problem with some of the places, that interact with procedures, where they might be not able to choose one implementation.

**Comment by lambdasapple** [comment]
More documentation, a website, variadic functions, ability to use external functions.

**Comment by Ellipse0934** [comment]
I agree with @datawater on more documentation. Let's start a github pages site which hosts [the manual](https://github.com/farkon00/cont/blob/6d89de144c53673d60efa235cf7bd59169326f70/learn-cont/00-learn-cont.md) and the quickstart guide.

Are you interested in this @farkon00  ? Should I open an issue to track this ?

**Comment by farkon00** [comment]
> I agree with @datawater on more documentation. Let's start a github pages site which hosts [the manual](https://github.com/farkon00/cont/blob/6d89de144c53673d60efa235cf7bd59169326f70/learn-cont/00-learn-cont.md) and the quickstart guide.
> 
> Are you interested in this @farkon00 ? Should I open an issue to track this ?

Yes, as I have previously communicated in private messages with you, I do think it's a good idea to work on documenting the project. Github pages sound like a good idea too. Would you like to start it out and find a good way to integrate a markdown renderer with the github pages CD?

Also regarding the CD, I would like to point out the existence of [the table of contents script](https://github.com/farkon00/cont/blob/master/scripts/learn_cont_00.py), I think it would be quite nice to integrate it with the MD renderer.

I will create the issue regrading the documentation.
